pkg/grpc/server.go

```diff
@@ -131,8 +131,10 @@ func (s *server) getMiddlewareOptions() []grpc_go.ServerOption {
 		grpc_go.StreamInterceptor(diag.TracingGRPCMiddlewareStream(s.tracingSpec)),
 		grpc_go.UnaryInterceptor(diag.TracingGRPCMiddlewareUnary(s.tracingSpec)))
 
-	s.logger.Infof("enabled metrics grpc middleware")
-	opts = append(opts, grpc_go.StatsHandler(diag.DefaultGRPCMonitoring.ServerStatsHandler))
+	if s.config.EnableMetrics {
```

Comment:
I thought there are more code to set statshandler in dapr. Instead of passing EnabledMetrics and checking it, we can add `enabled` property in https://github.com/dapr/dapr/blob/6c3e6967da842e35fe08cf4bda2b2db791d210fd/pkg/diagnostics/grpc_monitoring.go#L27

Then we can check enabled property like you did in service_monitoring.

https://github.com/dapr/dapr/blob/6c3e6967da842e35fe08cf4bda2b2db791d210fd/pkg/diagnostics/grpc_monitoring.go#L40

https://github.com/dapr/dapr/blob/6c3e6967da842e35fe08cf4bda2b2db791d210fd/pkg/diagnostics/grpc_monitoring.go#L44

Reply:
Ah we need to add the similar code to client stats handler.. - https://github.com/dapr/dapr/blob/6c3e6967da842e35fe08cf4bda2b2db791d210fd/pkg/diagnostics/grpc_monitoring.go#L49

Reply:
If I'm changing to this pattern, I might as well do the same for http too.

Reply:
Is this resolved?